tests/run_all_tests.py

```diff
@@ -1,236 +1,217 @@
-reduced_model_file ="model.hoc"
-frequency = 0
-manual_total_nsegs = -1
-synapse_file = Path + "origRandomSynapses-10000"
-voltage_file = Path + "voltage_vectors_for_unit_test.txt"
-write_unit_test_vectors = WriteVectors
-create_type = 'allen'
-celsius = 34
-print(Createtxt())
-os.system(Createtxt())
-print("Done Test 9, Allen V1 L5PC\n")
-
-### Test 10 model 47804508 (L1) from the Allen celltypes data base
-Path = 'TestsFiles/Test_10_Allen_47804508/'
-orig_morphology_file = Path + "reconstruction.swc"
-orig_model_file = Path + "AllenTemplate.hoc"
-reduced_model_file ="model.hoc"
-frequency = 0
-manual_total_nsegs = -1
-synapse_file = Path + "origRandomSynapses-10000"
-voltage_file = Path + "voltage_vectors_for_unit_test.txt"
-write_unit_test_vectors = WriteVectors
-create_type = 'allen'
-celsius = 34
-print(Createtxt())
-os.system(Createtxt())
-print("Done Test 10, Allen V1 L1\n")
-
-### Test 11 model Human L2/3 Cell from Eyal et al 2016
-Path = 'TestsFiles/Test_11_Human_L2_3_Eyal/'
-orig_morphology_file = Path + "2013_03_06_cell08_876_H41_05_Cell2.ASC"
-orig_model_file = Path + "model_0603_cell08_cm045.hoc"
-reduced_model_file ="model.hoc"
-frequency = 0
-manual_total_nsegs = -1
-synapse_file = Path + "origRandomSynapses-10000"
-voltage_file = Path + "voltage_vectors_for_unit_test.txt"
-write_unit_test_vectors = WriteVectors
-create_type = 'human'
-celsius = 37
-print(Createtxt())
-os.system(Createtxt())
-print("Done Test 11, Human L2/3\n")
-
-### Test 12 Tufted Pyramidal Cell (L6) Markram et al. Cell (2015) ---- 
-Path = 'TestsFiles/Test_12_TPC_L6_Markram/'
-orig_morphology_file = Path + "dend-tkb070125a3_ch1_cc2_b_hw_60x_1_axon-tkb060223b3_ch1_cc2_o_ps_60x_1_-_Clone_5.asc"
-orig_model_file = Path + "cADpyr231_L6_TPC_L1_44f2206f70.hoc"
-reduced_model_file ="model.hoc"
-frequency = 0
-manual_total_nsegs = -1
-synapse_file = Path + "synapses_location.txt"
-voltage_file = Path + "voltage_vectors_for_unit_test.txt"
-write_unit_test_vectors = WriteVectors
-create_type = 'bbpnew'
-celsius = 34
-print(Createtxt())
-os.system(Createtxt())
-print("Done Test 12, Tufted Pyramidal Cell (L6) Markram et al. Cell (2015)\n")
-
-### Test 13 Double Bouquet Cell (L4) Markram et al. Cell (2015) ---- 
-Path = 'TestsFiles/Test_13_DBC_L4_Markram/'
-orig_morphology_file = Path + "C140600C-I1_-_Clone_2.asc"
-orig_model_file = Path + "cNAC187_L4_DBC_23ffe29c8b.hoc"
-reduced_model_file ="model.hoc"
-frequency = 0
-manual_total_nsegs = -1
-synapse_file = Path + "synapses_locations.txt"
-voltage_file = Path + "voltage_vectors_for_unit_test.txt"
-write_unit_test_vectors = WriteVectors
-create_type  = 'bbpnew'
-celsius = 34
-print(Createtxt())
-os.system(Createtxt())
-print("Done Test 13, Double Bouquet Cell (L4) Markram et al. Cell (2015)\n")
 
+WRITE_UNIT_TEST_VECTORS = False
+PLOT_VOLTAGES = False
+
+BASE_PATH = os.path.abspath(os.path.dirname(__file__))
+TESTDATA_PATH = os.path.join(BASE_PATH, 'TestsFiles')
+
+
+def run_reduce(morphology_file,
```

Comment:
similar to `Createtxt()`, but raises an exception when there is a failure - allows for a test harness to catch the failures, instead of reading the log